sdk/download.go

```diff
@@ -269,11 +269,12 @@ func UpdateFile(file, url string, client *http.Client) error {
 	}
 	t.Close()
 	tempFile := t.Name()
-	defer os.Remove(tempFile)
+	os.Remove(tempFile) // let DownloadFile create it and use umask
```

Comment:
We don't have to use go's TempFile though; by creating and removing a tempfile here, we're already giving up its logic for ensuring the file doesn't exist. At this point, we might as well be generating a random string instead and using it.